avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -37,15 +38,15 @@
 import org.apache.avro.AvroTypeException;
 import org.apache.avro.Schema;
 import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericFixed;
+import org.apache.avro.generic.GenericRecord;
 import org.apache.avro.generic.IndexedRecord;
 import org.apache.avro.io.Decoder;
-import org.apache.avro.io.parsing.ResolvingGrammarGenerator;
-import org.apache.avro.io.parsing.Symbol;
+import com.linkedin.avro.fastserde.backport.ResolvingGrammarGenerator;
```

Comment:
Having a quick question:
Since for all Avro versions, we are going  to  use the backported classes from  Avro-1.8, so it is might be too advanced for the lower versions than 1.8, and maybe not good enough for higher avro versions than 1.8.
For example, some feature/symbol maybe ignored in vanilla avro-1.4, but could be supported well in the back ported parser/resolver, which will cause the inconsistency between fast-avro and regular avro.

I don't have  any concrete examples here, just think loudly.
Did you observe any inconsistency between fast-avro and vanilla avro after adopting the back ported parser/resolvers?

Reply:
Well, the reason why I backported it is that there were indeed some significant differences, particularly between 1.8 and 1.9. This part of the Avro code was significantly refactored, allegedly for performance reasons, and it totally broke our code-gen. My understanding is that as long as the binary format is compatible, then these classes should all work interchangeably. I could be wrong of course... not sure.

Reply:
ResolvingGrammarGenerator is mostly used to handle schema parsing/evolution (reader/writer schema), right?
For binary format consistency, I don't doubt it, but more about the schema parsing logic. 
Avro-1.8 could introduce some features, which is not available in Avro-1.4, which means Avro-1.4 ResolvingGrammarGenerator may skip/ignore some symbols, which might  be  added in modern Avro versions, such as in 1.7/1.8. 
Essentially, I am wondering about the inconsistency between Vanilla Avro ResolvingGrammarGenerator and the back ported ResolvingGrammarGenerator from Avro-1.8 when customers are using other Avro versions than 1.8.
Again, I don't have any proof, and it should be fine since all tests pass.